modules/features/player/src/main/java/au/com/shiftyjelly/pocketcasts/player/view/ShareFragment.kt

```diff
@@ -4,124 +4,264 @@ import android.os.Bundle
+                                null,
+                                requireContext(),
+                                SharePodcastHelper.ShareType.EPISODE,
+                                SourceView.PLAYER,
+                                analyticsTracker
+                            ).showShareDialogDirect()
+                        }
+                    },
+                    onShareCurrentPosition = {
+                        if (podcast != null && episode is PodcastEpisode) {
+                            SharePodcastHelper(
+                                podcast,
+                                episode,
+                                episode.playedUpTo,
+                                requireContext(),
+                                SharePodcastHelper.ShareType.CURRENT_TIME,
+                                SourceView.PLAYER,
+                                analyticsTracker
+                            ).showShareDialogDirect()
+                        }
+                    },
+                    onShareOpenFileIn = {
+                        if (podcast != null && episode is PodcastEpisode) {
+                            SharePodcastHelper(
+                                podcast,
+                                episode,
+                                episode.playedUpTo,
+                                requireContext(),
+                                SharePodcastHelper.ShareType.EPISODE_FILE,
+                                SourceView.PLAYER,
+                                analyticsTracker
+                            ).sendFile()
+                        }
+                    }
+                )
             }
-            close()
-        }
-        binding.buttonShareEpisode.setOnClickListener {
-            if (podcast != null && episode is PodcastEpisode) {
-                SharePodcastHelper(
-                    podcast,
-                    episode,
-                    null,
-                    requireContext(),
-                    ShareType.EPISODE,
-                    SourceView.PLAYER,
-                    analyticsTracker
-                ).showShareDialogDirect()
+            viewModel.playingEpisodeLive.observe(viewLifecycleOwner) { (_, backgroundColor) ->
+                applyColor(theme, backgroundColor)
             }
-            close()
         }
-        binding.buttonShareCurrentPosition.setOnClickListener {
-            if (podcast != null && episode is PodcastEpisode) {
-                SharePodcastHelper(
-                    podcast,
-                    episode,
-                    episode.playedUpTo,
-                    requireContext(),
-                    ShareType.CURRENT_TIME,
-                    SourceView.PLAYER,
-                    analyticsTracker
-                ).showShareDialogDirect()
+        (dialog as? BottomSheetDialog)?.behavior?.state = BottomSheetBehavior.STATE_EXPANDED
+    }
+}
+
+@Composable
+private fun ShareScreen(
+    podcast: Podcast?,
```

Comment:
As we are passing `Podcast` and `BaseEpisode` object to composable function, Should we mark these classes with @Stable for compose-runtime?